test-three/amy-menu/src/menu.tsx

```diff
@@ -0,0 +1,93 @@
+import { Component } from "react";
+import menuJson from "./menu-data.json";
+
+type MenuItemProps = {
+  name: string;
+};
+
+class MenuSubItem extends Component<MenuItemProps> {
+  render() {
+    return (
+      <li>
+        {this.props.name}
+      </li>
+    );
+  }
+}
+
+class MenuItem extends Component<MenuItemProps> {
+  render() {
+    return (
+      <ul>
+        {this.props.name}
```

Comment:
would this generate valid html 5?

Reply:
It would not. Unordered list elements should not contain text, they should only clarify that an unordered list is to be created. List item <li> elements would show information {this.props.name}. Thanks for pointing out.